### PR TITLE
Fix render on IE11

### PR DESCRIPTION
**Pull request by IdosApplitools**
The first render is failing on IE11 since the first step is setting the viewport size and the Ultra Fast Grid has no DOM snapshot to re-render the job. By navigating to the page first and then setting the viewport we see a successful execution.

Added another example that utilizes more hooks.

### CODE REVIEW

File: hooksExample.mjs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+// Puppeteer: launch browser
+const browser = await puppeteer.launch({ headless: false });
+const page = await browser.newPage();
+
+// Applitools: global variables
+let visualGridRunner = null;
+let eyes = null;
+
+// extend runner to initialize Eyes before all steps, take screenshot after each step, and close Eyes after all steps
```
Comment (jecfish) [inline hooksExample.mjs]:
```suggestion
// Extend runner to initialize Eyes before all steps, take screenshot after each step, and close Eyes after all steps
```

File: hooksExample.mjs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const recording = parse(JSON.parse(recordingText));
+
+// Puppeteer: Create a runner and execute the script
+const runner = await createRunner(recording, new Extension(browser, page, 7000));
+
+// Puppeteer: clean up
+await runner.run();
+await browser.close();
+
+// Manage tests across multiple Eyes instances
```
Comment (jecfish) [inline hooksExample.mjs]:
```suggestion
// Applitools: Manage tests across multiple Eyes instances
```

File: hooksExample.mjs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    console.log('done');
+
+    // Applitools: clean up
+    await eyes.closeAsync();
+    await eyes.abortAsync(); // abort if Eyes were not properly closed
+    console.log("Eyes closed");
+  }
+}
+
+// Puppeteer: Read the JSON user flow
```
Comment (jecfish) [inline hooksExample.mjs]:
```suggestion
// Puppeteer: read the JSON user flow
```

File: hooksExample.mjs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const recordingText = fs.readFileSync('./order-a-coffee.json', 'utf8');
+const recording = parse(JSON.parse(recordingText));
+
+// Puppeteer: Create a runner and execute the script
```
Comment (jecfish) [inline hooksExample.mjs]:
```suggestion
// Puppeteer: create a runner and execute the script
```

File: order-a-coffee.json [hunk 5]
```diff
@@ -1,15 +1,6 @@
 {
   "title": "order-a-coffee",
   "steps": [
-    {
```
Comment (jecfish) [inline order-a-coffee.json]: @IdosApplitools This file is exported from the Chrome's Recorder, we should not change anything here. We should make our code work for that.

Comment (IdosApplitools) [reply order-a-coffee.json]: @jecfish Using `visualGridOptions` and IEv2 we are now able to see a successful execution for IE11 browser as well.
The cause for that was the blank page that we see from the `setViewport` step

**Review by jecfish — APPROVED** [review]

**Pull request merged**